WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProduct.swift

```diff
@@ -7,32 +7,37 @@ struct AddProduct: View {
     ///
     @Binding var isPresented: Bool
 
+    /// View model to drive the view.
+    ///
+    @ObservedObject var viewModel: AddProductViewModel
+
     var body: some View {
         NavigationView {
-            ScrollView {
-                // TODO: Make the product list searchable
-                LazyVStack {
-                    // TODO: Add a product row for each non-variable product in the store
-                    let viewModel = ProductRowViewModel(id: 1,
-                                                        name: "Love Ficus",
-                                                        sku: "123456",
-                                                        price: "20",
-                                                        stockStatusKey: "instock",
-                                                        stockQuantity: 7,
-                                                        manageStock: true,
-                                                        canChangeQuantity: false) // Temporary view model with fake data
-                    ProductRow(viewModel: viewModel)
-                }
-                .padding()
-            }
-            .navigationTitle(Localization.title)
-            .navigationBarTitleDisplayMode(.inline)
-            .toolbar {
-                ToolbarItem(placement: .cancellationAction) {
-                    Button(Localization.close) {
-                        isPresented.toggle()
+            GeometryReader { geometry in
+                ScrollView {
+                    // TODO: Make the product list searchable
+                    LazyVStack {
+                        ForEach(viewModel.productRows) { viewModel in
+                            ProductRow(viewModel: viewModel)
+                        }
                     }
+                    .padding()
+                    .renderedIf(viewModel.productRows.isNotEmpty)
```

Comment:
Suggestion: You could define an enum here, something like
```swift
enum ProductSection {
    case list(productVM)
    case empty
    case loading
    case etc
}
```

That way it will be much easier to read how the view is rendered according to the view model and will allow you to unit test that logic!

Reply:
I agree! I actually added an enum for this, as I started to work on syncing products from remote but I didn't add it to this branch. If you don't mind I'll leave it as-is for now but replace this with the enum in the next PR that adds syncing.